### PR TITLE
fix problem with aot, changes in scripts

### PR DESCRIPTION
**Pull request by Sebaxp95**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ install:
 - ps: C:\projects\spectre\scripts\InstallNode.ps1
 
 build_script:
-- cmd: ng build -env=%CONFIGURATION% --aot=false >nul 2>&1
+- cmd: ng build -env=%CONFIGURATION% --target=production >nul 2>&1
```
Comment (gmrukwa) [inline appveyor.yml]: As above.

File: src/app/generic-form/generic-form.module.ts [hunk 2]
```diff
@@ -36,7 +40,15 @@ import { GenericFormComponent } from './generic-form/generic-form.component';
     BrowserModule,
     BrowserAnimationsModule,
     MaterialDesignFrameworkModule,
-    JsonSchemaFormModule.forRoot(MaterialDesignFrameworkModule),
+    {
```
Comment (gmrukwa) [inline src/app/generic-form/generic-form.module.ts]: Why is this expanded now?

File: src/app/generic-form/generic-form/generic-form.component.spec.ts [hunk 3]
```diff
@@ -84,7 +88,15 @@ describe('GenericFormComponent', () => {
         BrowserModule,
         BrowserAnimationsModule,
         MaterialDesignFrameworkModule,
-        JsonSchemaFormModule.forRoot(MaterialDesignFrameworkModule),
+        {
```
Comment (gmrukwa) [inline src/app/generic-form/generic-form/generic-form.component.spec.ts]: As above.

File: src/app/generic-form/generic-form.module.ts (hunk 2 above)
Comment (Sebaxp95) [reply src/app/generic-form/generic-form.module.ts]: https://github.com/dschnelldavis/angular2-json-schema-form/issues/189

**Comment by Sebaxp95** [comment]
AOT problem with JsonSchemaFormModule.forRoot(MaterialDesignFrameworkModule):
https://github.com/dschnelldavis/angular2-json-schema-form/issues/189#issuecomment-365971521

File: appveyor.yml [hunk 4]
```diff
@@ -26,7 +28,12 @@ install:
 - ps: C:\projects\spectre\scripts\InstallNode.ps1
 
 build_script:
-- cmd: ng build -env=%CONFIGURATION% --aot=false >nul 2>&1
+- cmd:
+    if(configuration == prod) {
+      ng build -env=%CONFIGURATION% prod_flag >nul 2>&1
```
Comment (gmrukwa) [inline appveyor.yml]: You've probably missed `%`. Since you are `if`ing this, you can just put `--target=production` here.

File: .travis.yml [hunk 5]
```diff
@@ -31,7 +32,12 @@ before_script:
 - sh -e /etc/init.d/xvfb start
 script:
 - ng test --single-run true --progress false --browsers Firefox
-- ng build -env=$CONFIGURATION
+- |
+  if [ $CONFIGURATION = "prod" ]; then
+      ng build -env=$CONFIGURATION PROD_FLAG
```
Comment (gmrukwa) [inline .travis.yml]: You've probably missed `$`. Similarly as in the `appveyor.yml`.

**Comment by Sebaxp95** [comment]
[Fix aot errors](https://trello.com/c/a09fzyuG/488-fix-aot-errors)

**Pull request merged**